packages/dropdowns.next/src/elements/menu/Menu.tsx

```diff
@@ -65,8 +68,11 @@ export const Menu = forwardRef<HTMLUListElement, IMenuProps>(
       onChange
     });
 
-    const triggerProps: IMenuButtonProps = {
-      ...getTriggerProps({ type: 'button' }),
+    const { onClick, onKeyDown, ...buttonProps } = _buttonProps;
+
+    const triggerProps: IButtonProps & { ref: RefObject<HTMLButtonElement> } = {
+      ...buttonProps,
```

Comment:
`isCompact` should spread before the new `buttonProps` so the consumer can manage the override.